tests/test_lightningd.py

```diff
@@ -5064,6 +5065,31 @@ def test_fundee_forget_funding_tx_unconfirmed(self):
         # fundee will also forget and disconnect from peer.
         assert len(l2.rpc.listpeers(l1.info['id'])['peers']) == 0
 
+    @pytest.mark.xfail(strict=True)
+    def test_reserve_enforcement(self):
+        """Channeld should disallow you spending into your reserve"""
+        l1, l2 = self.connect(may_reconnect=True)
+
+        self.fund_channel(l1, l2, 10**6)
+        # Pay 1000 satoshi to l2.
+        self.pay(l1, l2, 1000000)
+
+        l2.stop()
+        # Reach around into the db to reduce reserve to 0 so it will try to
+        # violate it.
+        db = sqlite3.connect(os.path.join(l2.daemon.lightning_dir, "lightningd.sqlite3"))
```

Comment:
I'd rather add a parameter to the `db_query` method to skip copying and work on the live DB in case we are sure the daemon isn't touching it.